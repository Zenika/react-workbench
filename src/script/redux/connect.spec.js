/* eslint-env jest */
jest.mock('./store', () => ({
  dispatch: jest.fn(),
  getState: () => ({ a: 'state' }),
}))

const { connect, connectAll } = require('./connect')
const store = require('./store')

describe('redux/connect', () => {
  const func = jest.fn((arg1, arg2) => (dispatch, getState) => ({ arg1, arg2, ...getState(), dispatch: dispatch('action') }))

  describe('connect', () => {
    it('should connect a function', () => {
      store.dispatch.mockReset()
      const connected = connect(func)

      expect(connected('first', 2)).toMatchSnapshot()
      expect(store.dispatch.mock).toMatchSnapshot()
    })
  })

  describe('connectAll', () => {
    it('should connect all fields', () => {
      store.dispatch.mockReset()
      const connected = connectAll({
        func1: func,
        func2: func,
      })

      expect(connected.func1('first', 2)).toMatchSnapshot()
      expect(store.dispatch.mock).toMatchSnapshot()
      expect(connected.func2(2, 'first')).toMatchSnapshot()
      expect(store.dispatch.mock).toMatchSnapshot()
    })
  })
})
