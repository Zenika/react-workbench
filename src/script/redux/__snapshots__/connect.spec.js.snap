// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux/connect connect should connect a function 1`] = `
Object {
  "a": "state",
  "arg1": "first",
  "arg2": 2,
  "dispatch": undefined,
}
`;

exports[`redux/connect connect should connect a function 2`] = `
Object {
  "calls": Array [
    Array [
      "action",
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;

exports[`redux/connect connectAll should connect all fields 1`] = `
Object {
  "a": "state",
  "arg1": "first",
  "arg2": 2,
  "dispatch": undefined,
}
`;

exports[`redux/connect connectAll should connect all fields 2`] = `
Object {
  "calls": Array [
    Array [
      "action",
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;

exports[`redux/connect connectAll should connect all fields 3`] = `
Object {
  "a": "state",
  "arg1": 2,
  "arg2": "first",
  "dispatch": undefined,
}
`;

exports[`redux/connect connectAll should connect all fields 4`] = `
Object {
  "calls": Array [
    Array [
      "action",
    ],
    Array [
      "action",
    ],
  ],
  "instances": Array [
    undefined,
    undefined,
  ],
}
`;
